eagle-xl/scripts/bh_masses.py

```diff
@@ -0,0 +1,116 @@
+"""
+Makes a BH dyn. mass vs. subgrid mass plot. Uses the swiftsimio library.
+"""
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+from swiftsimio import load
+
+from unyt import unyt_quantity
+from matplotlib.colors import LogNorm
+from matplotlib.animation import FuncAnimation
+
+mass_bounds = [6e3, 6e9]
+
+
+def get_data(filename):
+    """
+    Grabs the data (masses in Msun).
+    """
+
+    data = load(filename)
+
+    mass_sub = data.black_holes.subgrid_masses.to("Msun")
+    mass_dyn = data.black_holes.dynamical_masses.to("Msun")
+    mass_seed = unyt_quantity(
+        float(data.metadata.parameters["EAGLEAGN:subgrid_seed_mass_Msun"]), "Msun"
+    )
+    mass_gas = np.min(data.gas.masses.to("Msun"))
+
+    return mass_sub, mass_dyn, mass_seed, mass_gas
+
+
+def setup_axes(mass_bounds, number_of_simulations: int):
+    """
+    Creates the figure and axis object. Creates a grid of a x b subplots
+    that add up to at least number_of_simulations.
+    """
+
+    sqrt_number_of_simulations = np.sqrt(number_of_simulations)
+    horizontal_number = int(np.ceil(sqrt_number_of_simulations))
+    # Ensure >= number_of_simulations plots in a grid
+    vertical_number = int(np.ceil(number_of_simulations / horizontal_number))
+
+    fig, ax = plt.subplots(
+        vertical_number, horizontal_number, squeeze=True, sharex=True, sharey=True
+    )
+
+    ax = np.array([ax]) if number_of_simulations == 1 else ax
+
+    if horizontal_number * vertical_number > number_of_simulations:
+        for axis in ax.flat[number_of_simulations:]:
+            axis.axis("off")
+
+    # Set all valid on bottom row to have the horizontal axis label.
+    for axis in np.atleast_2d(ax)[:][-1]:
+        axis.set_xlabel("Black Hole Dynamical Masses $M_{\\rm dyn}$ [M$_\odot$]")
```

Comment:
These are quite long and will overlap significantly in the case where we have multiple simulations plotted. Can you do something like "BH Dyn. Masses $M_{\\rm dyn}$ [M$_\odot$]"?